launchpadstation/launchpadstation.ino

```diff
@@ -0,0 +1,296 @@
+
+                Arduino      Rocket
+                GND----------GND   (ground in)
+                A0-----------Ombilical 1 (Telemetry/FPV enable)
+                A1-----------Ombilical 2 (Calibration start)
+
+                Arduino      Ignition Relay
+                GND----------GND   (ground in)
+                A2-----------IN1   (Command pin)
+                5V-----------VCC   (5V in)
+                Connect the ignition circuit on Normally Open side
+
+                Arduino      FILL/VENT Relay (NB: jumper between VCC and JD-VCC)
+                GND----------GND   (ground in)
+                A3-----------IN2   (Command pin for relay 2)
+                A4-----------IN1   (Command pin for relay 1)
+                5V-----------VCC   (5V in)
+                Connect the solenoids on Normally Open side
+*/
+
+// Single wire ombilicals to the rocket
+#define PIN_OMBI_TM A0 // Write LOW to this pin to disable the Telemetry and FPV transmitters
+#define PIN_OMBI_CA A1 // Write LOW to this pin to start a sensor calibration on the rocket
+// Pins where the relays are connected
+#define PIN_RELAY_FIRE A2 // Write LOW to this pin to enable the ignition circuit
+#define PIN_RELAY_FILL A3 // Write LOW to this pin to open solenoid 1
+#define PIN_RELAY_VENT A4 // Write LOW to this pin to open solenoid 2
+
+#define CMD_FILL_START 0x61 // 'a'
+#define CMD_FILL_STOP  0x62 // 'b'
+#define CMD_VENT_START 0x63 // 'c'
+#define CMD_VENT_STOP  0x64 // 'd'
+#define CMD_ARM        0x65 // 'e'
+#define CMD_DISARM     0x66 // 'f'
+#define CMD_FIRE_START 0x67 // 'g'
+#define CMD_FIRE_STOP  0x68 // 'h'
+#define CMD_TM_ENABLE  0x41 // 'A'
+#define CMD_TM_DISABLE 0x42 // 'B'
+#define CMD_CA_TRIGGER 0x43 // 'C'
+
+#define REPLY_ACK  0x3D // '=' This is returned if the command is successfuly executed
+#define REPLY_NACK 0x21 // '!' This is returned if the command is not successfuly executed
+
+uint8_t command = 0x00;
+bool is_filling = false;
+bool is_venting = false;
+bool is_armed = false;
+
+void setup()
+{
+  pinMode(PIN_RELAY_FIRE, OUTPUT);
+  pinMode(PIN_RELAY_FILL, OUTPUT);
+  pinMode(PIN_RELAY_VENT, OUTPUT);
+  pinMode(PIN_OMBI_TM, OUTPUT);
+  pinMode(PIN_OMBI_CA, OUTPUT);
+  // Disable the ignition circuit
+  digitalWrite(PIN_RELAY_FIRE, HIGH);
+  // Close the solenoids
+  digitalWrite(PIN_RELAY_FILL, HIGH);
+  digitalWrite(PIN_RELAY_VENT, HIGH);
+  // Default state is HIGH for the ombilicals
+  digitalWrite(PIN_OMBI_TM, HIGH);
+  digitalWrite(PIN_OMBI_CA, HIGH);
+
+  init_communication();
+}
+
+void loop()
+{
+  read_byte(&command);
+
+  if (command)
+  {
+    switch (command)
+    {
+    case CMD_FILL_START:
+      start_filling();
+      break;
+    case CMD_FILL_STOP:
+      stop_filling();
+      break;
+    case CMD_VENT_START:
+      start_venting();
+      break;
+    case CMD_VENT_STOP:
+      stop_venting();
+      break;
+    case CMD_ARM:
+      arm();
+      break;
+    case CMD_DISARM:
+      disarm();
+      break;
```

Comment:
Do we need these?

Reply:
I'd like to keep this *arm* step. It's easy to only add a security level on the Ground Station side to prevent unwanted ignition but as this code can also be used directly from a serial link with no securities I think it's good to have a security level embedded in the LPS